tests/phpunit/integration/Core/Permissions/PermissionsTest.php

```diff
@@ -192,8 +194,108 @@ public function test_get_capabilities() {
 			Permissions::VIEW_DASHBOARD,
 			Permissions::VIEW_MODULE_DETAILS,
 			Permissions::MANAGE_OPTIONS,
+			Permissions::VIEW_SHARED_DASHBOARD,
 		);
 
 		$this->assertEqualSets( $capabilities, Permissions::get_capabilities() );
 	}
+
+	public function test_dashboard_sharing_capabilities() {
```

Comment:
This test is a bit on the unwieldy side and it would be nice to split into separate tests. However it's a bit tricky trying to determine the best approach and as a result, if it's worthwhile. I can see why you have left it as one lump.

Unfortunately, PHPUnit falls a bit short when it comes to organising tests, as it's not possible to group a set of tests and have a single setup method for that group (beyond grouping by a class, that is, but we don't want to add this test-specific setup to the class-wide setup method here).

I'd be inclined to simply refactor into a few private methods, e.g. -

```php
public function test_dashboard_sharing_capabilities() {
  // Setup

  $this->verify_view_shared_dashboard_capability( /* args */ );

  $this->verify_read_shared_module_data_capability( /* args */ );

  $this->verify_module_sharing_capabilities( /* args */ );

  // Then finish up running the tests from `$disable_feature();`
}
```

What do you think?

Reply:
Oh absolutely - if you need single line comments to explain your code, that code is generally better off in a well named method along with a doc block for it! 😄 However, within tests, we don't generally add doc blocks so I didn't bother. But the private methods would look nicer. So done!